### PR TITLE
[UseCase] Added SnackbarGetColorsUseCase & SnackbarGetButtonTypeUseCase

### PR DESCRIPTION
**Pull request by LouisBorleeAdevinta**

### CODE REVIEW

File: Sources/Core/Model/SnackbarIntent.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            variant: variant
+        )
+    }
+
+    /// Get the colors to apply on popovers from an intent
+    /// - Parameters:
+    ///   - theme: Spark theme
+    ///   - variant: The variant of the snackbar
+    /// - Returns: SnackbarColors
+    public func getColors(
```
Comment (robergro) [inline Sources/Core/Model/SnackbarIntent.swift:45]: Just a simple question : why we need that ?

File: Sources/Core/UseCase/SnackbarGetButtonTypeUseCase.swift [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        case .accent:
+            return self.getAccentButtonType(variant: variant)
+        case .surfaceInverse:
+            return self.getSurfaceInverseButtonType(variant: variant)
+        }
+    }
+
+    private func getSuccessButtonType(variant: SnackbarVariant) -> SnackbarButtonType {
+        let buttonIntent = ButtonIntent.success
+        let buttonVariant: ButtonVariant
```
Comment (robergro) [inline Sources/Core/UseCase/SnackbarGetButtonTypeUseCase.swift]: If you want, to limit the number of lines, you can : 
```swift
let buttonVariant: ButtonVariant = switch variant {
        case .filled: .filled
        case .tinted: .tinted
        }
```

OR

```swift
return .init(
    intent: ButtonIntent.alert,
    variant: switch variant {
        case .filled: .filled
        case .tinted: .tinted
        }
)
```

File: Sources/Core/Model/SnackbarIntent.swift (hunk 1 above)
Comment (robergro) [inline Sources/Core/Model/SnackbarIntent.swift:45]: Should be tested I think

File: Sources/Core/UseCase/SnackbarGetButtonTypeUseCase.swift (hunk 2 above)
Comment (LouisBorleeAdevinta) [reply Sources/Core/UseCase/SnackbarGetButtonTypeUseCase.swift]: Done

File: Sources/Core/Model/SnackbarIntent.swift (hunk 1 above)
Comment (LouisBorleeAdevinta) [reply Sources/Core/Model/SnackbarIntent.swift:45]: Done

**Review by robergro — APPROVED** [review]

**Pull request merged**